File: UserService/src/main/java/com/slavacom/userservice/config/RestClientConfig.java
package com.slavacom.userservice.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;

@Configuration
public class RestClientConfig {

    @Bean
    public RestClient restClient() {
        return RestClient.builder().build();
    }

    @Bean("authServiceRestClient")
    public RestClient authServiceRestClient(@Value("${auth-service.url:http://localhost:8081}") String authServiceUrl) {
        return RestClient.builder().baseUrl(authServiceUrl).build();
    }
}

File: TaskService/src/main/kotlin/com/slavacom/taskservice/config/RestClientConfig.kt
package com.slavacom.taskservice.config

import org.springframework.boot.web.client.RestClientCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RestClientConfig {

    @Bean
    fun restClientCustomizer(loggingInterceptor: LoggingClientHttpRequestInterceptor): RestClientCustomizer {
        return RestClientCustomizer { restClientBuilder ->
            restClientBuilder.requestInterceptor(loggingInterceptor)
        }
    }
}
